src/runtime-rs/crates/resource/src/volume/hugepage.rs

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{
+    collections::HashMap,
+    fs::File,
+    io::{BufRead, BufReader},
+};
+
+use agent::Storage;
+use anyhow::{anyhow, Context, Result};
+use byte_unit::Byte;
+use kata_sys_util::fs::get_base_name;
+use kata_types::mount::KATA_EPHEMERAL_VOLUME_TYPE;
+
+use crate::share_fs::EPHEMERAL_PATH;
+
+use super::{Volume, BIND};
+
+const PROC_MOUNTS_FILE: &str = "/proc/mounts";
+const HUGETLBFS: &str = "hugetlbfs";
+const NODEV: &str = "nodev";
+
+// container hugepage
+pub(crate) struct Hugepage {
+    storage: Option<Storage>,
+    mount: oci::Mount,
+}
+
+/// handle hugepage
+impl Hugepage {
+    pub(crate) fn new(
+        mount: &oci::Mount,
+        options_map: HashMap<Byte, u64>,
+        fs_options: Vec<String>,
+    ) -> Result<Self> {
+        // Create mount option string
+        let option = if let Some(page_size) = get_page_size(fs_options) {
+            let page_size =
+                Byte::from_str(page_size).context("failed to create Byte object from String")?;
+            options_map
+                .get(&page_size)
+                .map(|size| format!("pagesize={},size={}", page_size.get_bytes(), size))
+        } else {
+            None
+        }
+        .context("failed to get huge page options")?;
+        let base_name = get_base_name(mount.source.clone())?
+            .into_string()
+            .map_err(|e| anyhow!("{:?}", e))?;
```

Comment:
I think it would be better to explain what the error is here. Seems 'e' would be the original OsString, in which there is no error hint.

Reply:
fixed